src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -391,13 +391,9 @@ protected function getEnv(string $name)
             $localName = $name;
         }
 
-        if ($processors->has($prefix)) {
-            $processor = $processors->get($prefix);
-        } else {
-            $processor = new EnvVarProcessor($this);
-            if (false === $i) {
-                $prefix = '';
-            }
+        $processor = $processors->has($prefix) ? $processors->get($prefix) : new EnvVarProcessor($this);
+        if (false === $i && EnvVarProcessor::class === \get_class($processor)) {
```

Comment:
or instanceof I don't know?

Reply:
Because we implemented the empty string behavior only in our EnvVarProcessor class.

Reply:
Looking at this again, I think it's safer to add the condition :thinking:

Reply:
I'm not fan of hardcoding a behavior for EnvVarProcessor only. Can't we add some phpdoc in the interface to explain how a processor should behave instead?

Reply:
We could, but my point is that we added this special behavior on 5.4 as a bug fix :sweat_smile: Isn't it too late to change the behavior for every consumer?

Reply:
I think it's too much an edge case, nobody replaced the "string" processor :)

Reply:
I added a comment but I'm not convinced it's needed/understandable. I let you takeover :smile: